core/lib/resources/presence/presence_manager.js

```diff
@@ -119,11 +121,18 @@ PresenceManager.prototype.stampExpiration = function(message) {
 };
 
 PresenceManager.prototype.addClient = function(socketId, userId, userType,
-                                                      userData, callback) {
+                                                      userData, data, callback) {
+
+  // FIXME: Method signature is getting unmanageable.  
```

Comment:
Patrick is working on a Message abstraction that will end up collapsing the argument list. I suggest waiting.